haskell-src/Concordium/Wasm.hs

```diff
@@ -311,6 +334,20 @@ data WasmModule
     | WasmModuleV1 (WasmModuleV V1)
     deriving (Eq, Show)
 
+-- Implement `FromJSON` and `ToJSON` instances for `WasmModule`.
+$( deriveJSON
```

Comment:
How does the resulting JSON look for this type, I find it hard to derive in my head? :sweat_smile:

Reply:
```
{
    "energy": 200000,
    "expiryTime": "2024-05-14T14:23:18Z+00:00",
    "nonce": 38,
    "payload": {
        "mod": {
            "content": {
                "source": "0061736d010000000...................................616e736665724572726f720200"
            },
            "version": "wasmModuleV1"
        },
        "transactionType": "deployModule"
    },
    "signature": {
        "0": {
            "0": "a4ad6d482b1744c81c6223478c50ad2053d5add62c415948ef2f000275b7446eda183373e8a113dac8781b7d346a088873080957c3c5b9715a7b56929a49bf07"
        }
    },
    "signer": "4jxvYasaPncfmCFCLZCvuL5cZuvR5HAQezCHZH7ZA7AGsRYpix",
    "version": 1
}
```

Reply:
Not sure how much energy we should put into the readability of the JSON, but it would be nice if the expiryTime was readable.
I also think it would be nice to clean up the JSON representation, maybe:
```
"payload": {
  "module": {
    "version": "v1",
    "source": "..."
  }
  "transactionType": "deployModule"
}
```